pkg/gce-pd-csi-driver/controller.go

```diff
@@ -267,7 +267,7 @@ func (gceCS *GCEControllerServer) ControllerPublishVolume(ctx context.Context, r
 
 	volKey, err := common.VolumeIDToKey(volumeID)
 	if err != nil {
-		return nil, status.Error(codes.NotFound, fmt.Sprintf("Could not find volume with ID %v: %v", volumeID, err))
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("ControllerPublishVolume volume ID is invalid: %v", err))
 	}
 
 	volKey, err = gceCS.CloudProvider.RepairUnderspecifiedVolumeKey(ctx, volKey)
```

Comment:
Do we want to check gce error codes here? What if we had temporary issues with the cloud provider, and not that the disk doesn't actually exist.

Reply:
there's no real way to distinguish between "disk not exists in zone because it may or may not exist in another zone but is temporarily not showing up" vs "disk does not exist in any zone because it just doesn't exist" since `RepairUnderspecifiedVolumeKey` queries all the zones in the region for the disk

Reply:
you could check if the zone check fails because of "not found" error code

Reply:
right, but it may be `NotFound` is some subset of zones, and some other error in other zones. What do we do in that case?

Reply:
If we looped through all zones and found it, then return success.  Otherwise, if all the zones returned notFound, then return notFound.  Otherwise return error?

Reply:
I noticed that disk name is not unique across zones. How we can be sure the disk we found while looping through the zones is exactly the one we want since we don't pass zone information?

Reply:
we can't be sure, but I believe this is consistent with behavior of in-tree plugin. This codepath is only hit for migration - any disks managed natively through the CSI Driver have zone/region information encoded in their volume ID

Reply:
To be on the safe side, can we search all zones and if we have multiple matches than return error?

Reply:
In this issue https://github.com/kubernetes/kubernetes/pull/65198, @verult mentioned that csi driver solve this issue, how it is solved?

Reply:
When using the CSI Driver natively the region/zone information is encoded in the volume ID. This unspecified/repair case is **only** for CSI Migration, in which case we will continue to have the same issue as before.

Reply:
alright i've done the original repair suggestion in a seperate commit, PTAL